### PR TITLE
Issue 653 SAR implementation statement differential

### LINKED ISSUE #653: SAR: SSP Implementation Statement Differential - 4.5
**Extended Description**
- As an ASAP Developer, in order to provide FedRAMP reviewers easy confirmation that historic observations, I want to identify all historic observations.

**Acceptance Criteria**
- [x] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [x] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [x] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [x] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
- [x] observation with a type value of 'ssp-statement-issue' has a uuid that matches a related-observation/@observation-uuid in a finding.
- [x] The above finding also has a implementation-statument-uuid value that matches a UUID of a statement in the original SSP.
- [x] The observation of type 'ssp-statement-issue' must have a method of the value 'EXAMINE'.

**Definition of Done**
 - [x] Acceptance criteria met
 - [x] Unit test coverage of our code > 95%
 - [x] Automated code quality checks passed
 - [x] Security reviewed and reported
 - [x] Reviewed against plain language guidelines
 - [x] Code must be self-documenting
 - [x] No local tech debt
 - [x] Load/performance tests passed – needs to be created/automated
 - [x] Documentation updated
 - [x] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
closes #653 
* observation of type 'ssp-statement-issue' has method of 'EXAMINE'
* observation of type 'ssp-statement-issue has uuid that matches a finding/related-observation/observation-uuid.
* finding/related-observation/observation-uuid that matches a observation of type 'ssp-statement-issue' uuid has an implementation-statement-uuid value that matches a statement uuid value in the associated SSP.

### CODE REVIEW

File: src/validations/rules/sar.sch [hunk 1]
```diff
@@ -176,6 +176,101 @@
...
+                diagnostics="has-type-ssp-statement-issue-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.5"
+                id="has-type-ssp-statement-issue"
+                role="error"
+                test="
+                    if (oscal:type = 'ssp-statement-issue')
+                    then
+                        oscal:method = 'EXAMINE'
+                    else
+                        true()">An observation with a type of 'ssp-statement-issue' also has a method of 'EXAMINE'.</sch:assert>
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:238]:
```suggestion
                        true()">An observation with a type of 'ssp-statement-issue' must also have a method of 'EXAMINE'.</sch:assert>
```

File: src/validations/rules/sar.sch [hunk 2]
```diff
@@ -176,6 +176,101 @@
...
+                diagnostics="has-type-ssp-statement-issue-matches-related-observation-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.5"
+                id="has-type-ssp-statement-issue-matches-related-observation"
+                role="error"
+                test="
+                    if (oscal:type = 'ssp-statement-issue')
+                    then
+                        @uuid[. = $related-observations]
+                    else
+                        true()">An observation with a type of 'ssp-statement-issue' has a matching
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:250]:
```suggestion
                        true()">An observation with a type of 'ssp-statement-issue' must have a matching
```

File: src/validations/rules/sar.sch [hunk 3]
```diff
@@ -176,6 +176,101 @@
...
+                    if (oscal:type = 'ssp-statement-issue')
+                    then
+                        if (@uuid[. = $related-observations])
+                        then
+                            ../oscal:finding[oscal:related-observation/@observation-uuid = current()/@uuid]/oscal:implementation-statement-uuid[. = $ssp-statement-uuids]
+                        else
+                            true()
+                    else
+                        true()"
+                unit:override-xspec="both">The finding that has a related-observation/observation-uuid that matches an observation/@uuid, also has a implementation-statement-uuid value that
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:270]:
```suggestion
                unit:override-xspec="both">The finding that has a related-observation/observation-uuid that matches an observation/@uuid, must also have a implementation-statement-uuid value that 
```

**Review by danielnaab — CHANGES_REQUESTED** [review]

**Review by GaryGapinski — APPROVED** [review]
It appears that whoever was drafting SAR stuff did not take into account that an SSP undergoes change and there is no apparent place at which the SSP version at the time of assessment is captured.

**Review by danielnaab — APPROVED** [review]

**Pull request merged**